src/05-derivatives/02-common-data-types.md

```diff
@@ -0,0 +1,311 @@
+`desc` keyword is a general purpose field with freeform values. To distinguish
+between multiple different versions of processing for the same input data the
+`desc` keyword should be used. Note that even though `space` and `desc`
+are optional at least one of them needs to be defined to avoid name conflict
+with the raw file.
+
+Examples:
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-space-MNI305_bold.nii.gz
+            sub-001_task-rest_run-space-MNI305_bold.json
+```
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-1_desc-MC_bold.nii.gz
+            sub-001_task-rest_run-1_desc-MC_bold.json
+```
+
+```Text
+pipeline1/
+    sub-001/
+        func/
+            sub-001_task-rest_run-1_desc-fmriprep_bold.nii.gz
+            sub-001_task-rest_run-1_desc-fmriprep_bold.json
+```
+
+All REQUIRED metadata fields coming from a derivative file’s source file(s) MUST
+be propagated to the JSON description of the derivative unless the processing
+makes them invalid (e.g., if a source 4D image is averaged to create a single
+static volume, a SamplingFrequency property would no longer be relevant). In
+addition, all processed files include the following metadata JSON fields:
+
+| **Key name**  | **Description**                                                                        |
+| ------------- | -------------------------------------------------------------------------------------- |
+| SkullStripped | REQUIRED. Whether the volume was skull stripped (non-brain voxels set to zero) or not. |
+
+## Masks
+
+Template:
+
+```Text
+<pipeline_name>/
+    sub-<participant_label>/
+        func|anat|dwi/
+        <source_keywords>[_space-<space>][_type-<type>][_desc-<label>]_mask.nii.gz
+```
+
+A binary (1 - inside, 0 outside) mask in the space defined by `<space>`.
+By default (i.e., if no transformation has taken place) the value of
+`space` should be set to `orig`.
+
+JSON metadata fields:
+
+| **Key name** | **Description**                                                                                                                                |
+| ------------ | ---------------------------------------------------------------------------------------------------------------------------------------------- |
+| RawSources   | Same as defined in [Introduction](01-introduction.md), but elevated from OPTIONAL to REQUIRED                                                  |
+| Type         | RECOMMENDED. Short identifier of the mask. Reserved values: `Brain` - brain mask, `Lesion` - lesion mask, `Face` - face mask, `ROI` - ROI mask |
+
+Examples:
+
+```Text
+func_loc/
+    sub-001/
+        func/
+           sub-001_task-rest_run-1_space-MNI305_desc-PFC_mask.nii.gz
+           sub-001_task-rest_run-1_space-MNI305_desc-PFC_mask.json
+```
+
+```Text
+manual_masks/
+    sub-001/
+        anat/
+            sub-001_T1w_label-desc_mask.nii.gz
+            sub-001_T1w_label-desc_mask.json
```

Comment:
These look wrong, may be smth like
```suggestion
            sub-001_desc-PFC_mask.json
```
(assuming it is remaining in the subject anat space)?

Reply:
Is it relevant to have a `mod-` or `space-` indicating `T1w`?

Reply:
Thanks for catching this @yarikoptic. @effigies `mod` is not valid in this context. `space` is optional and used in the other example.